### PR TITLE
[Edge] client IDs in handle

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Remove global table for client IDs, and move it to edge handle.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c [hunk 1]
```diff
@@ -176,21 +200,13 @@ _close_all_connection_in_handle (nns_edge_handle_s * eh)
...
-    intptr_t client_id = (intptr_t) l->data;
-
-    conn_data = _nns_edge_get_conn (client_id);
-    _nns_edge_close_connection (conn_data->sink_conn);
-    _nns_edge_close_connection (conn_data->src_conn);
-
-    _nns_edge_remove_conn (client_id);
+  g_hash_table_iter_init (&iter, eh->conn_table);
+  while (g_hash_table_iter_next (&iter, &key, NULL)) {
+    _nns_edge_remove_conn (GPOINTER_TO_UINT (key));
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c:205]: Nested lock. Is it onk?

**Review by myungjoo — COMMENTED** [review]
Possible locking bugs in the untouched code at ```_nns_edge_add_conn```:

LOCK
get conn X
UNLOCK
...
LOCK
if X is not found, add X
UNLOCK
```
may have two X added simultaneously.

File: gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c (hunk 1 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c:204]: Thanks, I removed table-lock in this function.

File: gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c [hunk 2]
```diff
@@ -147,16 +165,24 @@ _nns_edge_add_conn (int64_t client_id)
...
+/**
+ * @brief Add connection info to edge handle. The edge handle itself should contain the connected IDs.
+ * @note This function should be called with handle lock.
+ */
+static void
+_add_connection_to_handle (nns_edge_handle_s * eh, int64_t client_id,
+    nns_edge_conn_data_s * conn_data)
+{
+  gpointer key = GUINT_TO_POINTER (client_id);
+
```
Comment (songgot) [inline gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c:182]: I wonder if this need to check coon_data if null or not.

Comment (songgot) [reply gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c:182]: Is client_id okay? thank you.

**Comment by jaeyun-jung** [comment]
PR updated, global table is useless so it was removed after discussion with @gichan-jang.

File: gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c (hunk 1 above)
Comment (myungjoo) [reply gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c]: Don't remove lock for this reason. You can use trylock.

**Review by gichan-jang — APPROVED** [review]
LGTM

**Review by anyj0527 — APPROVED** [review]
LGTM ✔

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**